blockchain/peer_test.go

```diff
@@ -0,0 +1,219 @@
+package blockchain
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	cmn "github.com/tendermint/tendermint/libs/common"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/p2p"
+)
+
+// check if peer timer is running or not (a running timer can be successfully stopped)
+// Note: it does stop the timer!
+func checkByStoppingPeerTimer(t *testing.T, peer *bpPeer, running bool) {
+	assert.NotPanics(t, func() {
+		stopped := peer.StopBlockResponseTimer()
+		if running {
+			assert.True(t, stopped)
+		} else {
+			assert.False(t, stopped)
+		}
+	})
+}
+
+func TestPeerResetMonitor(t *testing.T) {
+	peer := NewBPPeer(
+		p2p.ID(cmn.RandStr(12)), 10,
+		func(err error, _ p2p.ID) {},
+		nil)
+	peer.SetLogger(log.TestingLogger())
+	peer.resetMonitor()
```

Comment:
do we have to test this method separately from desc/incrPending functions? As far as I understand, we only call it there.